core/src/main/scala/kafka/log/remote/RemoteLogManager.scala

```diff
@@ -120,8 +120,9 @@ class RemoteLogManager(fetchLog: TopicPartition => Option[Log],
   private val poolSize = rlmConfig.remoteLogManagerThreadPoolSize
   private val rlmScheduledThreadPool = new RLMScheduledThreadPool(poolSize)
 
-  // topic ids that are received on leadership changes, this map is NEVER cleared
-  private val topicIds: mutable.Map[String, Uuid] = mutable.Map.empty
+  // topic ids that are received on leadership changes, this map is cleared on stop partitions
+  private val topicPartitionIds: concurrent.Map[TopicPartition, Uuid] = new ConcurrentHashMap[TopicPartition, Uuid]().asScala
```

Comment:
This seems a little wasteful. A single topic can have even thousands of partitions with replicas on a broker. We are duplicating the `topicId` UUID for each of these partitions.

Can we instead maintain a map of topicId to partitions, and keep the `topicIds` map? That way, when new partitions (leader or follower) are added, we can update the `topicIdToPartitionsMap`. When `stopPartitions`
 gets called with `delete = true`, we can remove from this map, and if the map becomes empty, we can remove from `topicIds`.

Reply:
The same UUID instance is reused for all the partitions. This approach lgtm.

Reply:
UUIDs are just references and this code is easy to understand.